Check qla

MaxNoe opened the pull request:
I found it easier to start from the current `run_all_day_long_branch` with the code from here: https://github.com/fact-project/plot_qla than to merge the diverged run_all_day_long and flare_alert branches

MaxNoe commented:
This now also contains a bugfix for the `is_user_awake_check` 
which lead to this error, when a user was awake:
```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.5/site-packages/custos/checks/__init__.py", line 83, in wrapped_check
    self.check(*args, **kwargs)
  File "/opt/conda/lib/python3.5/site-packages/shifthelper/checks.py", line 36, in check
    if all(f() for f in self.checklist):
  File "/opt/conda/lib/python3.5/site-packages/shifthelper/checks.py", line 36, in <genexpr>
    if all(f() for f in self.checklist):
  File "/opt/conda/lib/python3.5/site-packages/wrapt/wrappers.py", line 522, in __call__
    args, kwargs)
  File "/opt/conda/lib/python3.5/site-packages/shifthelper/debug_log_wrapper.py", line 9, in log_call_and_result
    result = wrapped(*args, **kwargs)
  File "/opt/conda/lib/python3.5/site-packages/shifthelper/conditions.py", line 260, in is_nobody_awake
    return whoisonshift() not in awake
  File "/opt/conda/lib/python3.5/site-packages/pandas/core/generic.py", line 831, in __hash__
    ' hashed'.format(self.__class__.__name__))
TypeError: 'Series' objects are mutable, thus they cannot be hashed
```

dneise commented on shifthelper/tools/__init__.py:
```diff
@@ -24,6 +25,15 @@
+    return date.year * 10000 + date.month * 100 + date.day
+
+
+def tonight_night_integer():
```
this exists in pyfact, we should not reproduce it

dneise commented on shifthelper/qla.py:
```diff
@@ -0,0 +1,186 @@
+database = create_db_connection()
+
+
+def dorner_binning(data, bin_width_minutes=20):
```
there is a function called `dorner_binning(data, bin_width_minutes=20)` in https://github.com/fact-project/plot_qla/blob/master/plot_qla.py, why is this reproduced here?

MaxNoe commented:
I was reluctant to add more, not easy to install dependencies.

MaxNoe commented:
I added pyfact as a dependency and removed the duplicated code.

This also contains an important bugfix now: not overriding `custos.Check.message` with something that has a different api. This lead to huge messages containing the docstring of str.
